### DISCUSSION TITLE
How does k3s do for write-endurance?

### DISCUSSION THREAD

**Discussion by victorhooi** [k3s-io/k3s#5317] [Q&A]
Hi,

Has anybody got much experience installing k3s on boot media with low write-endurance? (e.g. MicroSD/SD cards - like in a Raspberry Pi, or say, USB flash drives?)

In my case, I'd be looking at SLC-based USB flash drives:

https://www.swissbit.com/en/products/nand-flash-products/usb-products/

However, the endurance/performance is still probably not as good as say, your standard 2.5" SSD drive.

Are there any tips for installing k3s in these environments? Where do the majority of the writes to the boot media come from? Has anybody run say, a Raspberry Pi k3s cluster off MicroSD cards long-term?

(For the actual k8s volumes, I'd be looking to back these on network storage - NFS, SMB etc, or maybe use a separate dedicated disk on the nodes, and run Ceph across those).

**Comment by brandond** [comment]
Don't use etcd on SD or USB flash drives. Don't use NFS, SMB, or any other network filesystem for the k3s root filesystem; containerd does not support this. Use USB3 SSD/NVME if at all possible. If you must use SD or USB flash, make sure they are  of sufficient throughput to support your datastore ops + workload ops, and have enough excess space to wear-level properly.
